dartagnan/src/main/java/com/dat3m/dartagnan/program/event/core/Event.java

```diff
@@ -227,16 +226,19 @@ public <T> T accept(EventVisitor<T> visitor) {
 	// Encoding
 	// -----------------------------------------------------------------------------------------------------------------
 
-	public void initializeEncoding(SolverContext ctx) { }
-
 	public void runLocalAnalysis(Program program, Context context) { }
 
 	// This method needs to get overwritten for conditional events.
 	public boolean cfImpliesExec() {
 		return true;
 	}
 
-	public BooleanFormula encodeExec(EncodingContext ctx) {
-		return ctx.getBooleanFormulaManager().makeTrue();
+	/**
+	 * Conditions related to the execution variable of this event.
+	 *
+	 * @return Executions of the associated program must meet all those assumptions.
+	 */
+	public Collection<BooleanFormula> encodeExec(EncodingContext ctx) {
+		return List.of();
```

Comment:
You kept `BooleanFormula` for almost all methods, but this one returns a `Collection<BooleanFormula>`.
I think no (public) encoding function should return a collection of formulas if it already knows that those formulas must get conjoined.

Reply:
This is meant to be a optimization:  If `makeTrue()` is a native library call, then it does not have to be called thousands of times, just to be conjoined.  It would suffice to return an `Optional`, but `Collection` was just slightly easier to deal with.

Reply:
It is no native call though (`JavaSMT` already caches true and false) and even if it was, it wouldn't matter.
The amount of events we encode is trivial in comparison to the amount of edges we encode.